Bottom: 0841906224bb250847983ffd2a5d0bee11b39b28
Top:    b325ccfd975dd153285cfe5ba9f66efd16427780
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-13 14:47:35 -0700

Refresh of fork-connect

---

diff --git a/src/preload.c b/src/preload.c
index 33f3baf..2b115d3 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -184,7 +184,7 @@ static inline int fd_getd(int index)
 	return fdi ? fdi->fd : index;
 }
 
-static inline enum fd_state fd_gets(int index)
+static inline enum fd_fork_state fd_gets(int index)
 {
 	struct fd_info *fdi;
 
@@ -514,7 +514,7 @@ static void fork_active(int socket)
 	if ((ret != sizeof msg) || msg)
 		goto err2;
 
-	ret = rconnect(ret, &sin6, len);
+	ret = rconnect(ret, &addr, len);
 	if (ret)
 		goto err2;
 
@@ -574,7 +574,7 @@ static void fork_passive(int socket)
 		goto lclose;
 
 	msg = 0;
-	len = real.send(sfd, &msg, sizeof msg, MSG_NODELAY);
+	len = real.write(sfd, &msg, sizeof msgs);
 	if (len != sizeof msg)
 		goto lclose;
